### PR TITLE
Firmware changes for Rev5

### PR DESCRIPTION
**Pull request by sjkelly**
With these changes we seem to be able to get a lock on CGEN with rev5.

### CODE REVIEW

File: firmware/main.c [hunk 1]
```diff
@@ -354,24 +391,25 @@ static int xtrx_init(void)
...
+	board_revision = board_get_revision();
 
 	printf("\n");
 	printf("Getting Board Revision...\n");
 	printf("-------------------------\n");
-	printf("Rev%d.\n", board_get_revision());
+	printf("Rev%d.\n", board_revision);
+
+	printf("Intializing DAC configuration...\n");
+	if (board_revision == 5) {
```
Comment (Keno) [inline firmware/main.c:403]: Move this off into a separate function.

File: firmware/main.c [hunk 2]
```diff
@@ -354,24 +391,25 @@ static int xtrx_init(void)
...
+		printf("DAC is DAC60501.\n");
+		// Set Gain register
+		adr = 0x04;
+		i2c1_read(DAC60501_I2C_ADDR, adr, dat_2, 2, true);
```
Comment (Keno) [inline firmware/main.c:407]: This is unnecessary.

File: firmware/main.c [hunk 3]
```diff
@@ -354,24 +391,25 @@ static int xtrx_init(void)
...
+		adr = 0x04;
+		dat_2[0] = 0x01;
```
Comment (Keno) [inline firmware/main.c:409]: Just use a `uint16_t` here.

File: firmware/main.c [hunk 4]
```diff
@@ -354,24 +391,25 @@ static int xtrx_init(void)
...
+		dat_2[1] = 0x01;
+		i2c1_write(DAC60501_I2C_ADDR, adr, dat_2, 2);
```
Comment (Keno) [inline firmware/main.c:411]: Add a comment what this is doing (Enabling the reference divider).

File: firmware/main.c [hunk 5]
```diff
@@ -194,26 +235,22 @@ static void temp_test(void)
...
 	/* Rev4 is equipped with a MCP7525 */
 	if (board_revision == 4) {
 		value = value & 0xfff; /* 12-bit full range */
 		cmd    = (0b0000 << 4) | (value >> 8);
 		dat[0] = (value & 0xff);
 		i2c1_write(MPC4725_I2C_ADDR, cmd, dat, 1);
-	/* Rev5 is equipped with a LTC26X6 */
+	/* Rev5 is equipped with a DAC60501 */
 	} else {
-		value = value & 0xfff;       /* 12-bit full range */
```
Comment (Keno) [inline firmware/main.c:212]: Implement this for the DAC and make sure the vctcxo_test runs?

File: firmware/main.c [hunk 6]
```diff
@@ -260,15 +260,20 @@ static void vctcxo_dac_set(int value) {
...
 		dat[0] = (value & 0xff);
 		i2c1_write(MCP4725_I2C_ADDR, cmd, dat, 1);
 	/* Rev5 is equipped with a DAC60501 */
 	} else {
-
+		value = value << 4;
+		dat[0] = value & 0xff;
+		dat[1] = (value & 0xff00) >> 4;
+		cmd = 0x08;
+		i2c1_write(DAC60501_I2C_ADDR, cmd, dat, 1);
```
Comment (Keno) [inline firmware/main.c:276]:
```suggestion
		i2c1_write(DAC60501_I2C_ADDR, cmd, dat, 2);
```

File: firmware/main.c [hunk 7]
```diff
@@ -260,15 +260,20 @@ static void vctcxo_dac_set(int value) {
...
-		value = value & 0xfff; /* 12-bit full range */
 		cmd    = (0b0000 << 4) | (value >> 8);
 		dat[0] = (value & 0xff);
 		i2c1_write(MCP4725_I2C_ADDR, cmd, dat, 1);
 	/* Rev5 is equipped with a DAC60501 */
 	} else {
-
+		value = value << 4;
+		dat[0] = value & 0xff;
+		dat[1] = (value & 0xff00) >> 4;
```
Comment (Keno) [inline firmware/main.c:274]: What's the `>> 4` for?

File: firmware/main.c [hunk 8]
```diff
@@ -260,15 +260,20 @@ static void vctcxo_dac_set(int value) {
...
 	if (board_revision == 4) {
-		value = value & 0xfff; /* 12-bit full range */
 		cmd    = (0b0000 << 4) | (value >> 8);
 		dat[0] = (value & 0xff);
 		i2c1_write(MCP4725_I2C_ADDR, cmd, dat, 1);
 	/* Rev5 is equipped with a DAC60501 */
 	} else {
-
+		value = value << 4;
+		dat[0] = value & 0xff;
```
Comment (Keno) [inline firmware/main.c:273]: I think the endianness of this is wrong. The DAC60501 takes MSDB first over i2c.

File: firmware/main.c (hunk 7 above)
Comment (sjkelly) [reply firmware/main.c:274]: it should be 8...

**Pull request merged**